internal/metric/mock.go

```diff
@@ -0,0 +1,191 @@
+// Copyright The OpenTelemetry Authors
```

Comment:
intentional to copy these over rather than leaving them in otel package?

Reply:
They're internal to the otel package and thus can't be used here.  https://github.com/open-telemetry/opentelemetry-go/issues/872 could perhaps provide an exported solution, but one doesn't exist yet.